### DISCUSSION TITLE
Primary Key and Optional Relation CreateInput type problem

### DISCUSSION THREAD

**Discussion by shahafa** [prisma/prisma#11384] [Q&A]
Hello,

I'm struggling with optional relation and primary key.
I've created the following Prisma schema to define an App with optional prop where the prop is a relation to another table:
```
model App {
  id   String   @id
  name String
  prop AppProp? @relation(fields: [id], references: [id])
}

model AppProp {
  id       String @id
  someProp String

  App App[]
}
```

However the `AppCreateInput` type generated by Prisma is missing the `id` field and pretty useless when I try to create a new app
```
  export type AppCreateInput = {
    name: string
    prop?: AppPropCreateNestedOneWithoutAppInput
  }
```

**Comment by rahXephonz** [comment]
Your app prop doesnt have any relation inside App
I think you must adding relations in AppProps like App id with @relation(fields: [appId], references: [id])

**Reply by shahafa** [reply]
Thanks @ioofy
This is a one to many relation and I have the `prop AppProp? @relation(fields: [id], references: [id])` inside `App` and `App App[]` inside `AppProp` - so I'm not sure what I'm missing

**Reply by pantharshit00** [reply]
Since you are using primary keys here, they won't allow null values so relation can't be optional. You can store values in a new column.

**Reply by vladi-stevanovic** [reply]
👋 Hello @shahafa I'm going through pending GitHub Discussions to check if you need any help. Is this issue still open?
Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! :raised_hands:

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
